src/selectors/entities/channel_categories.ts

```diff
@@ -189,6 +189,76 @@ export function makeFilterAutoclosedDMs(getAutocloseCutoff = getDefaultAutoclose
     );
 }
 
+export function makeFilterAutoclosedDMs(): (state: GlobalState, channels: Channel[], categoryType: string) => Channel[] {
+    return createSelector(
+        (state: GlobalState, channels: Channel[]) => channels,
+        (state: GlobalState, channels: Channel[], categoryType: string) => categoryType,
+        getMyPreferences,
```

Comment:
Also, if you stop using the OPEN_TIME preferences, you can just select your new preference as a number to pass in here and make this recalculate a bit less often.